pyani_plus/public_cli.py

```diff
@@ -133,50 +135,82 @@ def check_fasta(fasta: Path) -> list[Path]:
     return fasta_names
 
 
-def run_snakemake_with_progress_bar(
+def progress_bar_via_db_comparisons(
+    database: Path, run_id: int, interval: float = 1.0
+) -> None:
+    """Show a progress bar based on monitoring the DB entries.
+
+    Will only self-terminate once all the run's comparisons are
+    in the database! Up to the caller to terminate it early.
+    """
+    session = db_orm.connect_to_db(database)
+    run = session.query(db_orm.Run).where(db_orm.Run.run_id == run_id).one()
+
+    already_done = run.comparisons().count()
+    total = run.genomes.count() ** 2 - already_done
+
+    done = 0
+    old_db_version = -1
+    with Progress(*progress_columns) as progress:
+        task = progress.add_task("Comparing pairs", total=total)
+        while done < total:
+            sleep(interval)
+            # Have there been any DB changes?
+            db_version = (
+                session.connection().execute(text("PRAGMA data_version;")).one()[0]
+            )
```

Comment:
Saw this fail on a cluster, ``sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) disk I/O error``, might need a retry?